lib/api/proposal.dart

```diff
@@ -0,0 +1,48 @@
+class Proposal {
+  final int pid;
+  final String title;
+  final String description;
+  final String startTime;
+  final String endTime;
+  // final DateTime startTime;
+  // final DateTime endTime;
+  final int uid;
+  final int numYes;
+  final int numNo;
+
+  Proposal({
+    this.pid,
+    this.title,
+    this.description,
+    this.startTime,
+    this.endTime,
+    this.uid,
+    this.numYes,
+    this.numNo,
+  });
+
+  factory Proposal.fromJson(Map<String, dynamic> json) {
+    return Proposal(
+      pid: json['pid'],
+      title: json['title'],
+      description: json['description'],
+      startTime: json['starttime'],
+      endTime: json['endtime'],
+      uid: json['uid'],
+      numYes: json['numyes'],
+      numNo: json['numno'],
+    );
+  }
+
+  Map toProposalMap() {
+    var map = new Map<String, dynamic>();
+    map['title'] = title;
+    map['description'] = description;
+    map['starttime'] = startTime;
+    map['endtime'] = endTime;
+    map['uid'] = uid;
+    print("$title $description $startTime $endTime $uid");
+    print("MAP $map");
+    return map;
+  }
```

Comment:
what is the use of this function?

Reply:
it resturns a JSON based on the proposal attributes that is needed for a POST request. In APIConnect, i take that JSON, encode it as a JSON response and send it out. I made these toMap() functions for proposal, forum post, events, comments, and a bunch others. You can see the article here: https://androidkt.com/http-post-request-in-flutter/ You can refactor those fxn names all to toMap() if you'd like, similar to the fromJSON fxns.